iroha/src/queue.rs

```diff
@@ -75,14 +75,19 @@ impl Queue {
                 .pending_tx_by_hash
                 .get(&transaction_hash)
                 .expect("Failed to get tx by hash.");
+
+            let mut bad_transaction = true;
+
             if !transaction.is_expired(self.transaction_time_to_live)
                 && !transaction.is_in_blockchain(world_state_view)
             {
                 if let Ok(signature_condition_passed) =
                     transaction.check_signature_condition(world_state_view)
                 {
-                    if is_leader {
-                        if signature_condition_passed {
+                    bad_transaction = false;
+
+                    match (is_leader, signature_condition_passed) {
+                        (false, _) | (true, true) => {
```

Comment:
Tuples of boolean variables are hard to read and easy to make a mistake.
Please replace with enumeration.

Reply:
Changed to `if !is_leader || signature_condition_passed`